packages/@react-spectrum/s2/src/Table.tsx

```diff
@@ -11,85 +11,1103 @@
+      headingHeight: DEFAULT_HEADER_HEIGHT[scale]
+    });
+  }, [overflowMode, density, scale]);
+
+  // Starts when the user selects resize from the menu, ends when resizing ends
+  // used to control the visibility of the resizer Nubbin
+  let [isInResizeMode, setIsInResizeMode] = useState(false);
+  let onResizeStart = useCallback((widths) => {
+    propsOnResizeStart?.(widths);
+  }, [propsOnResizeStart]);
+  let onResizeEnd = useCallback((widths) => {
+    setIsInResizeMode(false);
+    propsOnResizeEnd?.(widths);
+  }, [propsOnResizeEnd, setIsInResizeMode]);
+
+  let context = useMemo(() => ({
+    isQuiet,
+    density,
+    overflowMode,
+    loadingState,
+    isInResizeMode,
+    setIsInResizeMode,
+    scale,
+    layout
+  }), [isQuiet, density, overflowMode, loadingState, scale, layout, isInResizeMode, setIsInResizeMode]);
+
+  let isLoading = loadingState === 'loading' || loadingState === 'loadingMore';
+  let scrollRef = useRef(null);
+  let memoedLoadMoreProps = useMemo(() => ({
+    isLoading: isLoading,
+    onLoadMore
+  }), [isLoading, onLoadMore]);
+  useLoadMore(memoedLoadMoreProps, scrollRef);
+  let isCheckboxSelection = props.selectionMode === 'multiple' || props.selectionMode === 'single';
+
+  return (
+    <ResizableTableContainer
+      onResize={propsOnResize}
+      onResizeEnd={onResizeEnd}
+      onResizeStart={onResizeStart}
+      className={(UNSAFE_className || '') + mergeStyles(tableWrapper, styles)}
+      style={UNSAFE_style}>
+      <UNSTABLE_Virtualizer layout={layout}>
+        <InternalTableContext.Provider value={context}>
+          <RACTable
+            ref={scrollRef}
+            style={{WebkitTransform: 'translateZ(0)'}}
+            className={renderProps => table({
+              ...renderProps,
+              scale,
+              isCheckboxSelection,
+              isQuiet
+            })}
+            selectionBehavior="toggle"
+            onRowAction={onAction}
+            {...otherProps} />
+        </InternalTableContext.Provider>
+      </UNSTABLE_Virtualizer>
+    </ResizableTableContainer>
+  );
+}
+
+const centeredWrapper = style({
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  width: 'full',
+  height: 'full'
+});
+
+export interface TableBodyProps<T> extends Omit<RACTableBodyProps<T>, 'style' | 'className' | 'dependencies'> {}
+
+export function TableBody<T extends object>(props: TableBodyProps<T>) {
+  let {items, renderEmptyState, children} = props;
+  let tableVisualOptions = useContext(InternalTableContext);
+  let {loadingState} = tableVisualOptions;
+  let emptyRender;
+  let renderer = children;
+  let stringFormatter = useLocalizedStringFormatter(intlMessages, '@react-spectrum/s2');
+  let loadMoreSpinner = (
+    <UNSTABLE_TableLoadingIndicator className={style({height: 'full', width: 'full'})}>
+      <div className={centeredWrapper}>
+        <ProgressCircle
```

Comment:
Maybe make the load more spinner a bit smaller? It seems big relative to the row height at the moment IMO

Reply:
Hmm, I used `size="S"` instead but that seems too small IMO. Custom sizes aren't supported for ProgressCircle either 
![image](https://github.com/user-attachments/assets/f1b76242-8f2b-4170-9e7f-cd15e0cb8b9b)
FWIW the progress circle sizing is the same as it was in S1

Reply:
with compact, it is right up against the edges. I think the smaller one looks better 
<img width="375" alt="Screenshot 2024-08-29 at 5 44 13 PM" src="https://github.com/user-attachments/assets/2a5f428e-34de-4550-91f5-33b1ee356353">

Reply:
ah I see what you mean, I'll go ahead and change it to small then

Reply:
actually what do you think about this:
![image](https://github.com/user-attachments/assets/888393cd-0788-4842-91b1-a39f31897350)

I've made the loader height row larger (60px like in S1)

Reply:
Sure, I'll post in the room